### ISSUE TITLE
Error occurs when exiting the scene

### ISSUE THREAD

**Issue by figuremout** [gilzoide/godot-dockable-container#12]
Take the TestScene.tscn in samples folder for example, run the scene in godot and everything works fine, but after click the close button, Godot console prints below message:
![bug](https://user-images.githubusercontent.com/44429569/172779463-2a5aa619-b5b4-4fe5-8535-043e47794e8f.JPG)
Thanks for any help.

**Comment by gilzoide** [comment]
Hi @githubzjm, sorry about the delay.
Hmm, it is probably related to how the layout nodes are all Resources and reference each other. The output of Godot using `--verbose` makes it even more likely.

It's been a while that I wanted to refactor the Layout to be the only Resource subclass and store the whole tree using primitive data (like Dictionaries and Arrays), this could be another reason to do it already =P
It would be a breaking change, given saved layout Resources wouldn't work after the change... But I guess that's ok.

**Comment by githubzjm** [comment]
Thanks for your reply, this project is amazing and i am looking forward to this refactor.

**Comment by gilzoide** [comment]
FYI I started working on this refactor today, it is in the [feature/refactor-layout](https://github.com/gilzoide/godot-dockable-container/tree/feature/refactor-layout) branch =D
The new code introduced some bugs, so it still needs more work. But I've just tested running the project now and the "Resources still in use at exit" message disappeared, so it's a good sign! Just to mention, this new branch was built on top of the main branch, so it supports Godot 4.X only.
